PurchasesCoreSwift/Public/Purchases.swift

```diff
@@ -0,0 +1,1275 @@
+     * associated with the current `appUserID`. If the receipt is being used by an existing user, the current
+     * `appUserID` will be aliased together with the `appUserID` of the existing user.
+     *  Going forward, either `appUserID` will be able to reference the same user.
+     *
+     * You shouldn't use this method if you have your own account system. In that case "restoration" is provided
+     * by your app passing the same `appUserId` used to purchase originally.
+     *
+     * @note This may force your users to enter the App Store password so should only be performed on request of
+     * the user. Typically with a button in settings or near your purchase UI. Use syncPurchasesWithCompletionBlock
+     * if you need to restore transactions programmatically.
+     */
+    @objc func restoreTransactions(completionBlock completion: ReceivePurchaserInfoBlock? = nil) {
+        purchasesOrchestrator.restoreTransactions(completion: completion)
+    }
+
+    /**
+     * Computes whether or not a user is eligible for the introductory pricing period of a given product.
+     * You should use this method to determine whether or not you show the user the normal product price or
+     * the introductory price. This also applies to trials (trials are considered a type of introductory pricing).
+     *
+     * @note Subscription groups are automatically collected for determining eligibility. If RevenueCat can't
+     * definitively compute the eligibilty, most likely because of missing group information, it will return
+     * `RCIntroEligibilityStatusUnknown`. The best course of action on unknown status is to display the non-intro
+     * pricing, to not create a misleading situation. To avoid this, make sure you are testing with the latest
+     * version of iOS so that the subscription group can be collected by the SDK.
+     *
+     * @param productIdentifiers Array of product identifiers for which you want to compute eligibility
+     * @param receiveEligibility A block that receives a dictionary of product_id -> `RCIntroEligibility`.
+     */
+    @objc
+    func checkTrialOrIntroductoryPriceEligibility(_ productIdentifiers: [String],
+                                                  completionBlock receiveEligibility: @escaping ReceiveIntroEligibilityBlock) {
+        receiptFetcher.receiptData(refreshPolicy: .onlyIfEmpty) { maybeData in
+            if let data = maybeData {
+                if #available(iOS 12.0, macOS 10.14, macCatalyst 13.0, tvOS 12.0, watchOS 6.2, *) {
+                    self.modernEligibilityHandler(maybeReceiptData: data,
+                                                  productIdentifiers: productIdentifiers,
+                                                  completionBlock: receiveEligibility)
+                } else {
+                    self.backend.getIntroEligibility(appUserID: self.appUserID,
```

Comment:
i think this could be simplified to
`if let data = maybeData { 
	//modernhandler
} else {
	self.backend.getIntroEligibility(maybeData ?? Data())
}` ?

looks like the handling for both backend calls is the same?

Reply:
Yup!
✅